core/src/main/java/org/elasticsearch/plugins/InstallPluginCommand.java

```diff
@@ -239,13 +239,35 @@ private Path unzip(Path zip, Path pluginsDir) throws IOException {
         return target;
     }
 
+    /** Checks that the plugin contain a top-level folder titled `elasticsearch` and only one sub-folder titled after the name of the plugin. */
+    private Path findPluginRoot(Path root) throws Exception {
```

Comment:
@rjernst well yes, I started with that but the new structure is supposed to be "elasticsearch/plugin_name" and we don't know the plugin name yet (not until we found the plugin descriptor). I guess we could unzip just the elasticsearch part and keep findPluginRoot to ensure that the inner directory is titled correctly and then pass the new plugin root to the other plugin installation code, what do you think ?